charts/aws-fsx-openzfs-csi-driver/templates/controller-deployment.yaml

```diff
@@ -0,0 +1,237 @@
+  template:
+    metadata:
+      labels:
+        app: fsx-openzfs-csi-controller
+        {{- include "aws-fsx-openzfs-csi-driver.labels" . | nindent 8 }}
+        {{- if .Values.controller.podLabels }}
+        {{- toYaml .Values.controller.podLabels | nindent 8 }}
+        {{- end }}
+      {{- if .Values.controller.podAnnotations }}
+      annotations:
+        {{- toYaml .Values.controller.podAnnotations | nindent 8 }}
+      {{- end }}
+    spec:
+      nodeSelector:
+        kubernetes.io/os: linux
+            {{- with .Values.controller.nodeSelector }}
+            {{- toYaml . | nindent 8 }}
+            {{- end }}
+      serviceAccountName: {{ .Values.controller.serviceAccount.name }}
+      priorityClassName: {{ .Values.controller.priorityClassName }}
+          {{- with default .Values.controller.affinity }}
+      affinity:
+            {{- toYaml . | nindent 8 }}
+          {{- end }}
+      tolerations:
+            {{- with .Values.controller.tolerations }}
+            {{- toYaml . | nindent 8 }}
+            {{- end }}
+          {{- with .Values.controller.securityContext }}
+      securityContext:
+            {{- toYaml . | nindent 8 }}
+          {{- end }}
+          {{- with .Values.controller.initContainers }}
+      initContainers:
+            {{- toYaml . | nindent 8 }}
+          {{- end }}
+      containers:
+        - name: fsx-openzfs-plugin
+          image: {{ printf "%s:%s" .Values.image.repository (default (printf "v%s" .Chart.AppVersion) (toString .Values.image.tag)) }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args:
+            - --endpoint=$(CSI_ENDPOINT)
+          env:
+            - name: CSI_ENDPOINT
+              value: unix:///var/lib/csi/sockets/pluginproxy/csi.sock
+            - name: AWS_ACCESS_KEY_ID
+              valueFrom:
+                secretKeyRef:
+                  name: aws-secret
+                  key: key_id
+                  optional: true
+            - name: AWS_SECRET_ACCESS_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: aws-secret
+                  key: access_key
+                  optional: true
+            {{- with .Values.controller.region }}
+            - name: AWS_REGION
+              value: {{ . }}
+            {{- end }}
+          envFrom:
+            {{- with .Values.controller.envFrom }}
+            {{- . | toYaml | nindent 12 }}
+            {{- end }}
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/lib/csi/sockets/pluginproxy/
+          {{- with .Values.controller.volumeMounts }}
+          {{- toYaml . | nindent 12 }}
+          {{- end }}
+          ports:
+            - name: healthz
+              containerPort: 9808
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /healthz
+              port: healthz
+            initialDelaySeconds: 10
```

Comment:
Why did you opt to hard code these values? Do we not see users, or ourselves adjusting these values?

Reply:
These values are hardcoded because I'm following precedent from the EBS/Lustre CSI drivers haha. But looking more closely, these configure certain behaviors of the csi driver pods in the cluster that probably  shouldn't be controlled by the user. For example, the behavior of the liveness probe here on the controller-plugin should have consistent values that we know are safe. If we let the user set these values, they could misconfigure themselves fairly easily.

Reply:
Yea I agree. We should hardcode any values that users shouldn't adjust to prevent misconfigurations. We should also ensure users only have to adjust the values file when they want to set their deployment options. We don't want users having to search for a missing value.